python/neuroglancer/ingest/volumes/gcloudvolume.py

```diff
@@ -5,52 +5,43 @@
 import numpy as np
 from tqdm import tqdm
 
-import lib
-from lib import clamp, xyzrange, Vec3, Bbox, min2
 import neuroglancer
-from volumes import Volume
-from google.cloud import storage as gstorage
+import neuroglancer.ingest.lib as lib
+from neuroglancer.ingest.lib import clamp, xyzrange, Vec3, Bbox, min2
+from neuroglancer.ingest.volumes import Volume
+from neuroglancer.ingest.base import Storage
 
 class GCloudVolume(Volume):
-  def __init__(self, dataset_name, layer, mip, cache_files=True, use_ls=True):
-    super(self.__class__, self).__init__()
 
-    self.dataset_name = dataset_name
-    self.layer = layer
+  def __init__(self, dataset_name, layer_name, mip=0, cache_files=True, use_ls=True):
+    self._dataset_name = dataset_name
+    self._layer_name = layer_name
     self.mip = mip
-    
-    self.cache_files = cache_files
-    self.use_ls = use_ls
-
-    blob = self.__getInfoBlob()
-
-    self.info = json.loads(blob.download_as_string())
+    self._cache_files = cache_files
+    self._use_ls = use_ls
 
-  def __getInfoBlob(self):
-    client = gstorage.Client(project=lib.GCLOUD_PROJECT_NAME)
-    bucket = client.get_bucket(lib.GCLOUD_BUCKET_NAME)
+    self._storage = Storage(dataset_name, layer_name) 
+    self._get_info()
 
-    info_path = os.path.join(self.dataset_name, self.layer, 'info')
-    return gstorage.blob.Blob(info_path, bucket)
+  def _get_info(self):
+    self._info_blob =  self._storage.get_blob(
```

Comment:
Local use of gcloud might not have secret keys in the same manner as a cloud server. Storage shouldn't depend on it.